application/src/main/java/org/thingsboard/server/actors/ruleChain/RuleNodeActorMessageProcessor.java

```diff
@@ -121,23 +136,27 @@ public void onRuleToSelfMsg(RuleNodeToSelfMsg msg) throws Exception {
     }
 
     void onRuleChainToRuleNodeMsg(RuleChainToRuleNodeMsg msg) throws Exception {
-        msg.getMsg().getCallback().onProcessingStart(info);
-        checkComponentStateActive(msg.getMsg());
-        TbMsg tbMsg = msg.getMsg();
-        int ruleNodeCount = tbMsg.getAndIncrementRuleNodeCounter();
-        int maxRuleNodeExecutionsPerMessage = getTenantProfileConfiguration().getMaxRuleNodeExecsPerMessage();
-        if (maxRuleNodeExecutionsPerMessage == 0 || ruleNodeCount < maxRuleNodeExecutionsPerMessage) {
-            apiUsageClient.report(tenantId, tbMsg.getCustomerId(), ApiUsageRecordKey.RE_EXEC_COUNT);
-            if (ruleNode.isDebugMode()) {
-                systemContext.persistDebugInput(tenantId, entityId, msg.getMsg(), msg.getFromRelationType());
-            }
-            try {
-                tbNode.onMsg(msg.getCtx(), msg.getMsg());
-            } catch (Exception e) {
-                msg.getCtx().tellFailure(msg.getMsg(), e);
-            }
+        if (!isMySingletonNode()) {
+            putToQueue(msg.getMsg());
```

Comment:
Ok, we push to another partition and will stay there forever? Is there any fallback strategy to return to the regular partition after the singleton node? 
Why is it important?
- Some nodes may have a state for the partition (by originator entityId)
- SequentialByOriginator will lose the order

Reply:
Message will be forwarded to the previous partition 

![Screenshot from 2023-04-24 16-36-46](https://user-images.githubusercontent.com/56396344/234029620-ca70848b-cc88-4972-b092-c9b71921f85e.png)

Reply:
It would be nice to put couple java docs to lead in what the point is and where to expect the following message forward. For a first-time programmer, it will save a couple of days to maintain this code.